### PR TITLE
Dsindex tests

### PR DESCRIPTION
**Pull request by SergeyTsimfer**
Added tests for batchflow.dsindex.DatasetIndex

### CODE REVIEW

**Comment by roman-kh** [comment]
Make sure that test code is checked by pylint

File: batchflow/tests/datasetindex_test.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+			dsindex_big.calc_split(shares=[0.5, 0.5, 0.5])
+		with pytest.raises(ValueError):
+			dsindex_big.calc_split(shares=[0.5, 0.5, 0.5, 0.5]) 
+		with pytest.raises(ValueError):
+			DatasetIndex(2).calc_split(shares=[0.5, 0.5, 0.5])
+
+	def test_baseset_calc_split_correctness_1(self, dsindex_big):
+		assert sum(dsindex_big.calc_split()) == SIZE**3
+
+	@pytest.mark.skip(reason='default value is 1, not 0. line 94 of batchflow/base.py')
```
Comment (dpodvyaznikov) [inline batchflow/tests/datasetindex_test.py]: `xfail` mark is more suitable in this case. `skip` means that you expect your test to pass only if some conditions are met, `xfail` means that test is expected to fail anyway. See this [article](https://docs.pytest.org/en/latest/skipping.html).

Check `skip` marks below too.

File: batchflow/tests/datasetindex_test.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+	def test_shuffle_callable(self, dsindex_list):
+		left = dsindex_list._shuffle(shuffle=np.random.permutation) 
+		assert (left != np.arange(len(dsindex_list))).all()
+		assert set(left) == set(np.arange(len(dsindex_list)))
+
+
+	#Tests for batchflow.DatasetIndex.split() method
+	def test_split_correctness(self, dsindex_big):
+		dsindex_big.split(shares=[0.5, 0.5, 0], shuffle=True)
+		assert len(dsindex_big) == (len(dsindex_big.train) +\
```
Comment (dpodvyaznikov) [inline batchflow/tests/datasetindex_test.py]: Fix multiple line statements according to [PEP8](https://pep8.org/#code-lay-out).
Check multilines below too.

File: batchflow/tests/datasetindex_test.py [hunk 3]
```diff
@@ -1,221 +1,294 @@
...
+    right = range(length)
+    assert (left == right).all()
+
+def test_create_batch_type(all_indices):
+    """ Method 'create_batch' must be type-preserving. """
+    dsi = all_indices()
+    length = len(dsi)
+    assert isinstance(dsi.create_batch(range(length)), DatasetIndex)
+
+def test_next_batch_drop_last(all_indices):
```
Comment (roman-kh) [inline batchflow/tests/datasetindex_test.py]: More tests are needed for ``next_batch``: for all combinations of batch_size, n_epochs, drop_last

**Pull request merged**